src/exoticatechnologies/modifications/exotics/impl/ShipRepulsorSystem.kt

```diff
@@ -0,0 +1,586 @@
+            val radius: Float = getRadiusAmount(member, mods, exoticData)
+            val workMode = getWorkMode(member, mods, exoticData)
+
+            // Depending on the workmode, grab ships within radius with some prefiltering ...
+            val potentiallyAffectedShips = when (workMode) {
+                RepulsorWorkMode.ENEMIES_ONLY -> CombatUtils.getShipsWithinRange(ship.location, radius)
+                        // make sure it only contains enemies and not enemies and neutrals
+                        .filter { filterShip -> ship.owner != filterShip.owner && filterShip.owner != 100 }
+
+                RepulsorWorkMode.ENEMIES_AND_ALLIES -> CombatUtils.getShipsWithinRange(ship.location, radius)
+                        // make sure it only contains non-neutrals
+                        .filter { filterShip -> filterShip.owner != 100 }
+
+                RepulsorWorkMode.ALLIES_ONLY -> CombatUtils.getShipsWithinRange(ship.location, radius)
+                        // make sure it only contains allies and not enemies and neutrals
+                        .filter { filterShip -> filterShip.owner == ship.owner && filterShip.owner != 100 }
+            }.exhaustive
+                    // And ... then apply some more filtering
+                    // make sure we're not targetting ourselves
+                    .filter { module -> module.fleetMember != member && module.parentStation != ship && module != ship }
+                    // make sure we're not targetting child modules
+                    .filter { module -> module.parentStation == null }
+
+            return potentiallyAffectedShips
+        }
+
+        private fun showVisualFlair() {
+            // generate dots
+            val center = ship.location
+            val fullRange = getRadiusAmount(member, mods, exoticData)
+            // Lets draw the first ring at 1.5x collision radius so it's more visible, 1x is kinda "too close"
+            val stage1distance = ship.collisionRadius * 1.5f
+            val stage1DotsPair = generateDots(center, stage1distance)
+
+            val stage1left = stage1DotsPair.first
+            val stage1right = stage1DotsPair.second
+            // the first stage will draw emp arcs from left/right start to end
+            // start of stage1
+            for (index in 0 until stage1left.size -1) {
+                val fromL = stage1left[index]
+                val toL = stage1left[index+1]
+                Global
+                        .getCombatEngine()
+                        .spawnEmpArcVisual(
+                                fromL,
+                                ship,
+                                toL,
+                                ship,
+                                6f,
+                                Color.BLUE.darker().darker(),
+                                Color.WHITE
+                        )
```

Comment:
should be replaced with `EngineParticlePainter` even tho we will get rid of the whole EMP arc thing